// hmfull/HMfull.js
// Require HMtai Library //
const hmtai = require('hmtai')
// Require Fetch Functions //
const hmlinks = require('./fetchAPI.js')

module.exports = class HM {
    // Manual on what libraries return //
    static Manual = 'Example For All Libraries: HMfull.HMtai.sfw.wallpaper() - { url: "link" }';

    static HMtai = { // HMtai Library //
        sfw: {
            wallpaper() {
                return { url: hmtai.wallpaper() }
            },
            mobileWallpaper() {
                return { url: hmtai.mobileWallpaper() }
            },
            neko() {
                return { url: hmtai.neko() }
            },
            jahy() {
                return { url: hmtai.jahy() }
            }
        },
        nsfw: {
            ass() {
                return { url: hmtai.nsfw.ass() }
            },
            bdsm() {
                return hmtai.nsfw.bdsm()
            },
            blowjob() {
                return { url: hmtai.nsfw.blowjob() }
            },
            boobjob() {
                return { url: hmtai.nsfw.boobjob() }
            },
            creampie() {
                return { url: hmtai.nsfw.creampie() }
            },
            cum() {
                return { url: hmtai.nsfw.cum() }
            },
            vagina() {
                return { url: hmtai.nsfw.vagina() }
            },
            uniform() {
                return { url: hmtai.nsfw.uniform() }
            },
            manga() {
                return { url: hmtai.nsfw.manga() }
            },
            foot() {
                return { url: hmtai.nsfw.foot() }
            },
            femdom() {
                return { url: hmtai.nsfw.femdom() }
            },
            gangbang() {
                return { url: hmtai.nsfw.gangbang() }
            },
            hentai() {
                return { url: hmtai.nsfw.hentai() }
            },
            incest() {
                return { url: hmtai.nsfw.incest() }
            },
            ahegao() {
                return { url: hmtai.nsfw.ahegao() }
            },
            neko() {
                return { url: hmtai.nsfw.nsfwNeko() }
            },
            gif() {
                return { url: hmtai.nsfw.gif() }
            },
            ero() {
                return { url: hmtai.nsfw.ero() }
            },
            cuckold() {
                return { url: hmtai.nsfw.cuckold() }
            },
            orgy() {
                return { url: hmtai.nsfw.orgy() }
            },
            elves() {
                return { url: hmtai.nsfw.elves() }
            },
            pantsu() {
                return { url: hmtai.nsfw.pantsu() }
            },
            nsfwMobileWallpaper() {
                return { url: hmtai.nsfw.nsfwMobileWallpaper() }
            },
            glasses() {
                return { url: hmtai.nsfw.glasses() }
            },
            tentacles() {
                return { url: hmtai.nsfw.tentacles() }
            },
            thighs() {
                return { url: hmtai.nsfw.thighs() }
            },
            yuri() {
                return { url: hmtai.nsfw.yuri() }
            },
            zettaiRyouiki() {
                return { url: hmtai.nsfw.zettaiRyouiki() }
            },
            masturbation() {
                return { url: hmtai.nsfw.masturbation() }
            },
            public() {
                return { url: hmtai.nsfw.public() }
            }
        }
    }

    static Nekos = { // Nekos REST API //
        ball8() {
            return hmlinks.nekosGet('8ball')
        },
        sfw: {
            slap() {
                return hmlinks.nekosGet('slap')
            },
            poke() {
                return hmlinks.nekosGet('poke')
            },
            pat() {
                return hmlinks.nekosGet('pat')
            },
            neko() {
                return hmlinks.nekosGet('neko')
            },
            kiss() {
                return hmlinks.nekosGet('kiss')
            },
            hug() {
                return hmlinks.nekosGet('hug')
            },
            feed() {
                return hmlinks.nekosGet('feed')
            },
            cuddle() {
                return hmlinks.nekosGet('cuddle')
            },
            smug() {
                return hmlinks.nekosGet('smug')
            },
            baka() {
                return hmlinks.nekosGet('baka')
            },
            tickle() {
                return hmlinks.nekosGet('tickle')
            },
            nekogif() {
                return hmlinks.nekosGet('ngif')
            },
            kemonomimi() {
                return hmlinks.nekosGet('kemonomimi')
            },
            holo() {
                return hmlinks.nekosGet('holo')
            },
        },
        nsfw: {
            hentai() {
                return hmlinks.nekosGet('Random_hentai_gif')
            },
            neko() {
                return hmlinks.nekosGet('nsfw_neko_gif')
            },
            lesbian() {
                return hmlinks.nekosGet('les')
            },
            kuni() {
                return hmlinks.nekosGet('kuni')
            },
            classic() {
                return hmlinks.nekosGet('classic')
            },
            boobs() {
                return hmlinks.nekosGet('boobs')
            },
            blowjob() {
                return hmlinks.nekosGet('blowjob')
            },
            anal() {
                return hmlinks.nekosGet('anal')
            },
            yuri() {
                return hmlinks.nekosGet('yuri')
            },
            trap() {
                return hmlinks.nekosGet('trap')
            },
            tits() {
                return hmlinks.nekosGet('tits')
            },
            femdom() {
                return hmlinks.nekosGet('femdom')
            },
            feet() {
                return hmlinks.nekosGet('feetg')
            },
            ero() {
                return hmlinks.nekosGet('ero')
            },
            pussy() {
                return hmlinks.nekosGet('pussy')
            },
            erofeet() {
                return hmlinks.nekosGet('erofeet')
            },
            erokitsu() {
                return hmlinks.nekosGet('erok')
            },
            lewdkitsu() {
                return hmlinks.nekosGet('lewdk')
            },
            lewdneko() {
                return hmlinks.nekosGet('lewd')
            },
            solog() {
                return hmlinks.nekosGet('solog')
            },
            bj() {
                return hmlinks.nekosGet('bj')
            },
            pwankg() {
                return hmlinks.nekosGet('pwankg')
            },
            erokemo() {
                return hmlinks.nekosGet('erokemo')
            },
            cum() {
                return hmlinks.nekosGet('cum')
            },
            keta() {
                return hmlinks.nekosGet('keta')
            },
            spank() {
                return hmlinks.nekosGet('spank')
            },
            eroyuri() {
                return hmlinks.nekosGet('eroyuri')
            },
            eroneko() {
                return hmlinks.nekosGet('eron')
            },
            lewdkemo() {
                return hmlinks.nekosGet('lewdkemo')
            },
            pussyJpg() {
                return hmlinks.nekosGet('pussy_jpg')
            },
            cumJpg() {
                return hmlinks.nekosGet('cum_jpg')
            },
            avatar() {
                return hmlinks.nekosGet('nsfw_avatar')
            },
            wallpaper() {
                return hmlinks.nekosGet('wallpaper')
            },
            waifu() {
                return hmlinks.nekosGet('waifu')
            },
            holoero() {
                return hmlinks.nekosGet('holoero')
            },
            hololewd() {
                return hmlinks.nekosGet('hololewd')
            },
            futarani() {
                return hmlinks.nekosGet('futarani')
            },
            gasm() {
                return hmlinks.nekosGet('gasm')
            },
            foxgirl() {
                return hmlinks.nekosGet('foxgirl')
            }
        }
    }

    static NekoLove = { // Neko-Love REST API //
        sfw: {
            pat() {
                return hmlinks.nekoloveGet('pat')
            },
            hug() {
                return hmlinks.nekoloveGet('hug')
            },
            kiss() {
                return hmlinks.nekoloveGet('kiss')
            },
            cry() {
                return hmlinks.nekoloveGet('cry')
            },
            slap() {
                return hmlinks.nekoloveGet('slap')
            },
            smug() {
                return hmlinks.nekoloveGet('smug')
            },
            punch() {
                return hmlinks.nekoloveGet('punch')
            },
            neko() {
                return hmlinks.nekoloveGet('neko')
            },
            kitsune() {
                return hmlinks.nekoloveGet('kitsune')
            },
            waifu() {
                return hmlinks.nekoloveGet('waifu')
            }
        },
        nsfw: {
            nekolewd() {
                return hmlinks.nekoloveGet('nekolewd')
            }
        }
    }

    static Miss = { // Miss REST API //
        meme() {
            return hmlinks.missGet('meme')
        },
        sfw: {
            hug() {
                return hmlinks.missGet('hug')
            },
            kiss() {
                return hmlinks.missGet('kiss')
            },
            cry() {
                return hmlinks.missGet('cry')
            },
            kill() {
                return hmlinks.missGet('kill')
            },
            view() {
                return hmlinks.missGet('view')
            },
            angry() {
                return hmlinks.missGet('angry')
            },
            dance() {
                return hmlinks.missGet('dance')
            }
        },
        nsfw: {
            ero() {
                return hmlinks.missGet('hentai')
            },
            pussy() {
                return hmlinks.missGet('pussy')
            },
            boobs() {
                return hmlinks.missGet('boobs')
            }
        }
    }

    static Freaker = { // ComputerFreaker REST API //
        sfw: {
            baguette() {
                return hmlinks.freakerGet('baguette')
            },
            dva() {
                return hmlinks.freakerGet('dva')
            },
            neko() {
                return hmlinks.freakerGet('neko')
            },
            yuri() {
                return hmlinks.freakerGet('yuri')
            },
            anime() {
                return hmlinks.freakerGet('anime')
            }
        },
        nsfw: {
            hentai() {
                return hmlinks.freakerGet('hentai')
            },
            neko() {
                return hmlinks.freakerGet('nsfwneko')
            },
            trap() {
                return hmlinks.freakerGet('trap')
            }
        }
    }
}
