// hanime-api/dist/src/util/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = {
    SEARCH_URL: 'https://search.htv-services.com/',
    VIDEO_URL: 'https://hw.hanime.tv/api/v8/video?id=',
    SITE_URL: 'https://hanime.tv/videos/hentai/',
    BROWSE_URL: 'https://hw.hanime.tv/api/v8/browse/',
    LANDING_URL: 'https://hw.hanime.tv/api/v8/landing/',
    STORYBOARDS_URL: 'https://hanime.tv/rapi-cache/rapi/v7/hentai_video_storyboards?hv_id=',
    SEASONS_URL: 'https://hr.hanime.tv/api/v8/hentai_videos?source=browse-seasons&__order=',
    TAGS: [
        '3d',
        'ahegao',
        'anal',
        'bdsm',
        'big boobs',
        'blow job',
        'bondage',
        'boob job',
        'censored',
        'comedy',
        'cosplay',
        'creampie',
        'dark skin',
        'facial',
        'fantasy',
        'filmed',
        'foot job',
        'futanari',
        'gangbang',
        'glasses',
        'hand job',
        'harem',
        'hd',
        'horror',
        'incest',
        'inflation',
        'lactation',
        'loli',
        'maid',
        'masturbation',
        'milf',
        'mind break',
        'mind control',
        'monster',
        'nekomimi',
        'ntr',
        'nurse',
        'orgy',
        'plot',
        'pov',
        'pregnant',
        'public sex',
        'rape',
        'reverse rape',
        'rimjob',
        'scat',
        'school girl',
        'shota',
        'softcore',
        'swimsuit',
        'teacher',
        'tentacle',
        'threesome',
        'toys',
        'trap',
        'tsundere',
        'ugly bastard',
        'uncensored',
        'vanilla',
        'virgin',
        'watersports',
        'x-ray',
        'yaoi',
        'yuri',
    ],
};
