// hanime-api/dist/src/structures/PartialVideo.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartialVideo = void 0;
const util_1 = require("../util");
const constants_1 = __importDefault(require("../util/constants"));
const toOmit = ['description', 'rating', 'poster_url', 'cover_url', 'brand_id', 'duration_in_ms', 'is_censored', 'monthly_rank', 'created_at', 'released_at'];
class PartialVideo {
    /** @ignore */
    constructor(client, raw) {
        Object.defineProperties(this, {
            client: { value: client },
            raw: { value: raw },
        });
        const filteredRaw = Object.assign({}, raw);
        for (const k of toOmit)
            delete filteredRaw[k];
        Object.assign(this, filteredRaw);
        this.description = raw.description.replace(/(\n|\s{2,}|<\/?\w+>)/g, '');
        this.url = constants_1.default.SITE_URL + raw.slug;
        this.poster = util_1.getImageURL(raw.poster_url);
        this.cover = util_1.getImageURL(raw.cover_url);
        this.brandID = raw.brand_id;
        this.duration = raw.duration_in_ms;
        this.censored = raw.is_censored;
        this.monthlyRank = raw.monthly_rank;
        this.createdAt = new Date(raw.created_at * 1000);
        this.createdTimestamp = raw.created_at * 1000;
        this.releasedAt = new Date(raw.released_at * 1000);
        this.releasedTimestamp = raw.released_at * 1000;
    }
    /**
     * Fetch full video data.
     */
    fetch() {
        return this.client.fetchVideo(this);
    }
    /**
     * Fetch the storyboards of this video.
     */
    fetchStoryboards() {
        return this.client.fetchStoryboards(this);
    }
}
exports.PartialVideo = PartialVideo;
