// hanime-api/dist/src/structures/PartialFetchedVideo.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartialFetchedVideo = void 0;
const util_1 = require("../util");
const constants_1 = __importDefault(require("../util/constants"));
const PartialTag_1 = require("./PartialTag");
const toOmit = ['description', 'tags', 'is_visible', 'is_hard_subtitled', 'is_banned_in', 'created_at_unix', 'released_at_unix', 'rating', 'preview_url', 'primary_color', 'hentai_tags', 'poster_url', 'cover_url', 'brand_id', 'duration_in_ms', 'is_censored', 'monthly_rank', 'created_at', 'released_at'];
/**
 * Organized data of a video accessed via another video.
 */
class PartialFetchedVideo {
    /** @ignore */
    constructor(client, raw) {
        var _a, _b, _c;
        Object.defineProperties(this, {
            client: { value: client },
            raw: { value: raw },
        });
        const filteredRaw = Object.assign({}, raw);
        for (const k of toOmit)
            delete filteredRaw[k];
        Object.assign(this, filteredRaw);
        this.description = (_a = raw.description) === null || _a === void 0 ? void 0 : _a.replace(/(\n|\s{2,}|<\/?\w+>)/g, '');
        this.url = constants_1.default.SITE_URL + raw.slug;
        this.poster = util_1.getImageURL(raw.poster_url);
        this.cover = util_1.getImageURL(raw.cover_url);
        this.brandID = +raw.brand_id;
        this.duration = raw.duration_in_ms;
        this.censored = raw.is_censored;
        this.monthlyRank = raw.monthly_rank;
        this.createdAt = new Date(raw.created_at_unix * 1000);
        this.createdTimestamp = raw.created_at_unix * 1000;
        this.releasedAt = new Date(raw.released_at_unix * 1000);
        this.releasedTimestamp = raw.released_at_unix * 1000;
        this.visible = !!raw.is_visible;
        this.hardSubtitled = raw.is_hard_subtitled;
        this.bannedIn = ((_b = raw.is_banned_in) === null || _b === void 0 ? void 0 : _b.split(',')) || [];
        this.tags = ((_c = raw.hentai_tags) === null || _c === void 0 ? void 0 : _c.map(x => new PartialTag_1.PartialTag(client, x))) || [];
    }
    /**
     * Fetch full video data.
     */
    fetch() {
        return this.client.fetchVideo(this);
    }
    /**
     * Fetch the storyboards of this video.
     */
    fetchStoryboards() {
        return this.client.fetchStoryboards(this);
    }
}
exports.PartialFetchedVideo = PartialFetchedVideo;
