// hanime-api/dist/src/structures/Video.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Video = void 0;
const PartialFetchedVideo_1 = require("./PartialFetchedVideo");
const Tag_1 = require("./Tag");
const Franchise_1 = require("./Franchise");
const Brand_1 = require("./Brand");
const Storyboard_1 = require("./Storyboard");
const Server_1 = require("./Server");
/**
 * Organized data of a video.
 */
class Video {
    /** @ignore */
    constructor(client, raw) {
        Object.defineProperties(this, {
            client: { value: client },
            raw: { value: raw },
        });
        this.data = new PartialFetchedVideo_1.PartialFetchedVideo(client, raw.hentai_video);
        this.tags = raw.hentai_tags.map(d => new Tag_1.Tag(client, d));
        this.franchise = new Franchise_1.Franchise(client, raw.hentai_franchise, raw.hentai_franchise_hentai_videos);
        this.storyboards = raw.hentai_video_storyboards.map(d => new Storyboard_1.Storyboard(d));
        this.brand = new Brand_1.Brand(client, raw.brand);
        this.nextVideo = new PartialFetchedVideo_1.PartialFetchedVideo(client, raw.next_hentai_video);
        this.nextRandomVideo = new PartialFetchedVideo_1.PartialFetchedVideo(client, raw.next_random_hentai_video);
        this.manifest = {
            servers: raw.videos_manifest.servers.map(d => new Server_1.Server(d)),
        };
        this.client.storyboards.set(`${this.data.id}`, this.storyboards);
        this.client.storyboards.set(this.data.slug, this.storyboards);
    }
}
exports.Video = Video;
