// lewds.api/src/end.js
module.exports = {
    base: "https://api.lewds.fun/api/nsfw/",
    baseFun: "https://api.lewds.fun/fun/",
    baseSafe: "https://api.lewds.fun/sfw/",
    endpoints: {
        ass: "https://api.lewds.fun/api/nsfw/ass",
        futa: "https://api.lewds.fun/api/nsfw/futa",
        furgif: "https://api.lewds.fun/api/nsfw/furgif",
	furfuta: "https://api.lewds.fun/api/nsfw/furfuta",
        threed: "https://api.lewds.fun/api/nsfw/3d",
        boobs: "https://api.lewds.fun/api/nsfw/boobs",
        athighs: "https://api.lewds.fun/api/nsfw/athighs",
        feet: "https://api.lewds.fun/api/nsfw/feet",
        gifs: "https://api.lewds.fun/api/nsfw/gifs",
	hboobs: "https://api.lewds.fun/api/nsfw/hboobs",
	hentai: "https://api.lewds.fun/api/nsfw/hentai",
	kink: "https://api.lewds.fun/api/nsfw/kink",
	thighs: "https://api.lewds.fun/api/nsfw/thighs",
	yuri: "https://api.lewds.fun/api/nsfw/yuri",
	milk: "https://api.lewds.fun/api/nsfw/milk",
	slime: "https://api.lewds.fun/api/nsfw/slime",
	trap: "https://api.lewds.fun/api/nsfw/trap",
	kiss: "https://api.lewds.fun/api/sfw/kiss",
	pantsu: "https://api.lewds.fun/api/nsfw/pantsu",
	pat: "https://api.lewds.fun/api/fun/pat",
	eightball: "https://api.lewds.fun/api/fun/8ball",
	reverse: "https://api.lewds.fun/api/fun/reverse",
	shuffle: "https://api.lewds.fun/api/fun/shuffle",
    }
}
